compiler/rustc_ast/src/util/classify.rs

```diff
@@ -247,7 +247,7 @@ fn type_trailing_braced_mac_call(mut ty: &ast::Ty) -> Option<&ast::MacCall> {
                 break (mac.args.delim == Delimiter::Brace).then_some(mac);
             }
 
-            ast::TyKind::Ptr(mut_ty) | ast::TyKind::Ref(_, mut_ty) => {
+            ast::TyKind::Ptr(mut_ty) | ast::TyKind::Ref(_, _, mut_ty) => {
```

Comment:
I'd strongly prefer if we split this out into a totally different `TyKind`. These are not refs, and I think the last time we've shoved new options into a preexisting type (`dyn*`) it led to more headaches than expected.

Reply:
That makes good sense to me. I'll work on that.

Reply:
Okay, I've broken out `PinnedRef` as a separate variant. In most cases I'm still treating them the same, but that's more obvious in the source now.